sstables/sstables.cc

```diff
@@ -1406,6 +1410,66 @@ void sstable::write_filter() {
     write_simple<component_type::Filter>(filter_ref);
 }
 
+void sstable::maybe_rebuild_filter_from_index(uint64_t num_partitions) {
+    if (!has_component(component_type::Filter)) {
+        return;
+    }
+
+    // Skip rebuilding the bloom filter if the false positive rate based
+    // on the current bitset size is within 75% to 125% of the configured
+    // false positive rate.
```

Comment:
How much work do we need for the filter rebuild in terms of calculation and disk io? If it is cheap, perhaps we can drop the 25% threshold. 

Since we can rebuild the filter, maybe we can stop the estimating at repair and streaming level to save the work.  @denesb @raphaelsc

Reply:
@asias In terms of work, we read all the partition keys from the index file, calculate the hashes and set the bloom filter based on that. In my tests it took anywhere between 0-5% of the time taken for the compaction depending on the number of keys and number of writes happening in the table. (I have provided more statistics of such a manual test in the PR description).